### PR TITLE
:bug: Enable alff to run without nuisance

### PR DESCRIPTION
**Pull request by e-kenneally**
## Description
The current alff/fAlff workflow is set up so that if nuisance regression is in native space and alff is in template space, CPAC runs a nodeblock called `warp_denoiseNofilt_to_T1template`. 

From cpac_pipeline.py: 
Screenshot 2023-11-02 at 2 33 02 PM
From registration.py: 
Screenshot 2023-11-02 at 2 32 05 PM
This nodeblock requires a file from nuisance regression (`desc-denoisedNofilt_bold`), so it crashes if nuisance regression is not enabled. Since we don't want alff and nuisance regression to be inter-dependent, I moved the transform nodes from their separate transform nodeblock into the template-space alff nodeblock. Now the transform step gets triggered only if `desc-denoisedNofilt_bold` is in the strat_pool. 

## Tests
I successfully ran fmriprep ingress + alff with and without nuisance regression.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository. 
- [x] My commit messages follow best practices.
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I updated the changelog.
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: CPAC/alff/alff.py [hunk 1]
```diff
@@ -314,9 +353,10 @@ def alff_falff_space_template(wf, cfg, strat_pool, pipe_num, opt=None):
     node, out = strat_pool.get_data(["space-template_res-derivative_desc-bold_mask",
                                      "space-template_desc-bold_mask"])
     wf.connect(node, out, alff, 'inputspec.rest_mask')
-    outputs = {
+    
+    outputs = outputs.update({
```
Comment (shnizzedy) [inline CPAC/alff/alff.py:357]: `dict().update` works in-place
```suggestion
    outputs.update({
```

**Review by shnizzedy — APPROVED** [review]
<span title="Nice">[n(o)ice](https://en.wikipedia.org/wiki/Brooklyn_Nine-Nine)</span>

Is it true that `desc-denoisedNofilt_bold` only exists `if 'Template' in target_space_alff and 'Native' in target_space_nuis`, or is just checking for the existence `desc-denoisedNofilt_bold` a better condition for what we care about here?

**Comment by e-kenneally** [comment]
> Is it true that `desc-denoisedNofilt_bold` only exists `if 'Template' in target_space_alff and 'Native' in target_space_nuis`, or is just checking for the existence `desc-denoisedNofilt_bold` a better condition for what we care about here?

`desc-denoisedNofilt_bold` only exists if nuisance regression is running (in native or template space). I made that the check because regardless of the space for nuisance regression, if the file isn't being produced then the template space transform step can't run.

**Comment by shnizzedy** [comment]
> `desc-denoisedNofilt_bold` only exists if nuisance regression is running (in native or template space). I made that the check because regardless of the space for nuisance regression, if the file isn't being produced then the template space transform step can't run.

Makes sense. I was just wondering if `desc-denoisedNofilt_bold` will exist / we need to apply the transform if nuisance and ALFF are both in template space?

**Comment by e-kenneally** [comment]
> Makes sense. I was just wondering if `desc-denoisedNofilt_bold` will exist / we need to apply the transform if nuisance and ALFF are both in the same space?

Ohh I see! There's a template space version of that file too, so if nuisance runs in template space this won't be triggered. The transform is now only within the template space alff nodeblock, so the conditions that would trigger the transform step are still native nuisance + template alff

**Review by shnizzedy — APPROVED** [review]
I [already approved](https://github.com/FCP-INDI/C-PAC/pull/2015#pullrequestreview-1716025073), but I'm approving extra because now I understand better

**Review by sgiavasis — APPROVED** [review]

**Pull request merged**